### ISSUE TITLE
 Athena doesn't distinguish between `SELECT ` and `DDLs` and fails incorrectly w/ "unparsable section"

### ISSUE THREAD

**Issue by chollinger93** [sqlfluff/sqlfluff#5312]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff` + `Athena` fails to parse the following query:

```sql
-- CREATE OR REPLACE VIEW "foobar" AS 
select b.start,
cast(b.foo as date) as bar 
from dbt.baz b 
where b.START = '2020-01-01'
```

Due to the `start` keyword incorrectly being seen as "reserved". See below.

### Expected Behaviour

 `start` is **not** a reserved keyword for a `SELECT` statement as per the spec: https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html

That's **only** true for DDLs.

```sql
-- CREATE OR REPLACE VIEW "foobar" AS 
select dd.`start`, cast(dd.foo as date) as bar
from dbt.baz dd
where dd.`start` = '2020-01-01'
```

Is accepted. Both are valid in the Athena UI.

I thought the "CREATE OR REPLACE VIEW" (commented out) had something to do with it, but that renders a `inline_comment` token, as expected.

### Observed Behaviour

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    inline_comment:                                           '-- CREATE OR REPLACE VIEW "foobar" AS '
[L:  1, P: 39]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    whitespace:                                               ' '
[L:  2, P:  2]      |    statement:
[L:  2, P:  2]      |        select_statement:
[L:  2, P:  2]      |            select_clause:
[L:  2, P:  2]      |                keyword:                                      'select'
[L:  2, P:  8]      |                [META] indent:
[L:  2, P:  8]      |                whitespace:                                   ' '
[L:  2, P:  9]      |                select_clause_element:
[L:  2, P:  9]      |                    column_reference:
[L:  2, P:  9]      |                        naked_identifier:                     'b'
[L:  2, P: 10]      |                unparsable:                                   !! Expected: 'Nothing...'
[L:  2, P: 10]      |                    raw:                                      '.'
[L:  2, P: 11]      |                    code:                                     'start'
[L:  2, P: 16]      |                    comma:                                    ','
[L:  2, P: 17]      |                    whitespace:                               ' '
[L:  2, P: 18]      |                    code:                                     'cast'
[L:  2, P: 22]      |                    bracketed:
[L:  2, P: 22]      |                        start_bracket:                        '('
[L:  2, P: 23]      |                        code:                                 'b'
[L:  2, P: 24]      |                        raw:                                  '.'
[L:  2, P: 25]      |                        code:                                 'foo'
[L:  2, P: 28]      |                        whitespace:                           ' '
[L:  2, P: 29]      |                        code:                                 'as'
[L:  2, P: 31]      |                        whitespace:                           ' '
[L:  2, P: 32]      |                        code:                                 'date'
[L:  2, P: 36]      |                        end_bracket:                          ')'
[L:  2, P: 37]      |                    whitespace:                               ' '
[L:  2, P: 38]      |                    code:                                     'as'
[L:  2, P: 40]      |                    whitespace:                               ' '
[L:  2, P: 41]      |                    code:                                     'bar'
[L:  2, P: 44]      |            whitespace:                                       ' '
[L:  2, P: 45]      |            [META] dedent:
[L:  2, P: 45]      |            from_clause:
[L:  2, P: 45]      |                keyword:                                      'from'
[L:  2, P: 49]      |                whitespace:                                   ' '
[L:  2, P: 50]      |                from_expression:
[L:  2, P: 50]      |                    [META] indent:
[L:  2, P: 50]      |                    from_expression_element:
[L:  2, P: 50]      |                        table_expression:
[L:  2, P: 50]      |                            table_reference:
[L:  2, P: 50]      |                                naked_identifier:             'dbt'
[L:  2, P: 53]      |                                dot:                          '.'
[L:  2, P: 54]      |                                naked_identifier:             'baz'
[L:  2, P: 57]      |                        whitespace:                           ' '
[L:  2, P: 58]      |                        alias_expression:
[L:  2, P: 58]      |                            [META] indent:
[L:  2, P: 58]      |                            naked_identifier:                 'b'
[L:  2, P: 59]      |                            [META] dedent:
[L:  2, P: 59]      |                    [META] dedent:
[L:  2, P: 59]      |                whitespace:                                   ' '
[L:  2, P: 60]      |            where_clause:
[L:  2, P: 60]      |                keyword:                                      'where'
[L:  2, P: 65]      |                [META] (implicit) indent:
[L:  2, P: 65]      |                whitespace:                                   ' '
[L:  2, P: 66]      |                expression:
[L:  2, P: 66]      |                    column_reference:
[L:  2, P: 66]      |                        naked_identifier:                     'b'
[L:  2, P: 67]      |                [META] dedent:
[L:  2, P: 67]      |            unparsable:                                       !! Expected: 'Nothing...'
[L:  2, P: 67]      |                raw:                                          '.'
[L:  2, P: 68]      |                code:                                         'START'
[L:  2, P: 73]      |                whitespace:                                   ' '
[L:  2, P: 74]      |                raw:                                          '='
[L:  2, P: 75]      |                whitespace:                                   ' '
[L:  2, P: 76]      |                numeric_literal:                              '2020'
[L:  2, P: 80]      |                raw:                                          '-'
[L:  2, P: 81]      |                numeric_literal:                              '01'
[L:  2, P: 83]      |                raw:                                          '-'
[L:  2, P: 84]      |                numeric_literal:                              '01'
[L:  2, P: 86]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   2 | P:  10 |  PRS | Line 2, Position 10: Found unparsable section: '.start, cast(b.foo
                       | as date) as bar'
L:   2 | P:  67 |  PRS | Line 2, Position 67: Found unparsable section: '.START =
                       | 2020-01-01'
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
```

### How to reproduce

```bash
echo '-- CREATE OR REPLACE VIEW "foobar" AS \n select b.start, cast(b.foo as date) as bar from dbt.baz b where b.START = '2020-01-01'' > sqlfluff_invalid_1.sql
echo 'select b.start, cast(b.foo as date) as bar from dbt.baz b where b.START = '2020-01-01'' > sqlfluff_invalid_2.sql
echo 'select b.`start`, cast(b.foo as date) as bar from dbt.baz b where b.`START` = '2020-01-01'' > sqlfluff_valid_2.sql
for f in ./sqlfluff*.sql; do echo "\n\n$f\n"; sqlfluff parse --dialect=athena $f; done
```

### Dialect

`athena`

### Version

```bash
sqlfluff --version
# sqlfluff, version 2.3.2
python --version
# Python 3.10.11
```

### Configuration

Vanilla

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

(Id love to, but I'm not sure if I have time and lexers/parsers/compilers aren't exactly my area of expertise)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by heuermh** [comment]
Ran into a similar error
```
$ cat athena-start-column.sql
SELECT
    start,
    "end"
FROM
    athena_table

$ sqlfluff lint athena-start-column.sql
== [athena-start-column.sql] FAIL
L:   2 | P:   1 | LT02 | Line should not be indented.
                       | [layout.indent]
L:   2 | P:   5 |  PRS | Line 2, Position 5: Found unparsable section: 'start,\n
                       | "end"'
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
```

Per
https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html#list-of-reserved-words-sql-select

reserved words must be enclosed in double quotes ("), however this appears to cause sqlfluff to interpret the identifiers as values
```
$ cat athena-start-column.sql
SELECT
    "start",
    "end"
FROM
    athena_table

$ sqlfluff lint athena-start-column.sql
== [athena-start-column.sql] FAIL
L:   2 | P:   5 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
L:   3 | P:   5 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
```


- [x] I also agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Most likely `start` is a keyword in Athena, which causes the parsing problem. Minimal repro:

```sql
select table1.start
from table1
```
